### DISCUSSION TITLE
Analytics combined with getserversideprops

### DISCUSSION THREAD

**Discussion by kevinsimper** [vercel/next.js#28993] [Help]
I want to log all our requests on the backend, and with nextjs and next/link a server request is made to two different urls depending on if it is first load or the user already is on a page.

So example:
file is called /src/pages/hello.js
first http request is /hello
if user is already on / and navigates via next/link then the http request to getserversideprops is
/_next/data/XXXX/hello.json

I don't want to track on the frontend as those requests can be blocked by the client.

I also looked in getServersideprops and it does not look like the request url is normalized, so context.req.url is different and could be used, but for analytics I want to se /hello and /_next/data/XXXX/hello.json as the same two events.

Have other people tackled this? Is this something that Next.js is missing?

**Comment by wilsonneto-swe** [comment]
You can just use a custom server and track all the requests you want with original routes.
I use this approach for my observability, logs, and traces. (but I use next router events for my google analytics page view tracs)

You can read more here:
https://nextjs.org/docs/advanced-features/custom-server

I hope it helps you.
I wish you all the success in your projects.
Best regards.
